src/main/java/de/chojo/repbot/service/GdprService.java

```diff
@@ -1,139 +1,75 @@
-        User user;
-        try {
-            user = shardManager.retrieveUserById(userId).complete();
-        } catch (RuntimeException e) {
-            log.info("Could not process gdpr request for user {}. User could not be retrieved.", userId);
-            return false;
-        }
-        if (user == null) {
-            log.info("Could not process gdpr request for user {}. User could not be retrieved.", userId);
-            return false;
-        }
-
-        PrivateChannel privateChannel;
-        try {
-            privateChannel = user.openPrivateChannel().complete();
-        } catch (RuntimeException e) {
-            log.info("Could not process gdpr request for user {}. Could not open private channel.", userId);
-            return false;
-        }
-        if (privateChannel == null) {
-            log.info("Could not process gdpr request for user {}. Could not open private channel.", userId);
-            return false;
-        }
-
-        var userData = gdprData.getUserData(user);
-
-        if (userData.isEmpty()) {
-            log.warn("Could not process gdpr request for user {}. Data aggregation failed.", userId);
-            return false;
-        }
-        Path tempFile;
-        try {
-            tempFile = Files.createTempFile("repbot_gdpr", ".json");
-        } catch (IOException e) {
-            log.warn("Coult not create temp file", e);
-            return false;
-        }
-        try {
-            Files.writeString(tempFile, userData.get());
-        } catch (IOException e) {
-            log.warn("Could not write to temp file", e);
-            return false;
-        }
-
-        try {
-            privateChannel
-                    .sendMessage("Here is your requested data. You can request it again in 30 days.")
-                    .addFile(tempFile.toFile())
-                    .complete();
-        } catch (RuntimeException e) {
-            log.warn("Could not send gdpr data to user {}. File sending failed.", userId, e);
-            return false;
-        }
-        return true;
-    }
 
     public CompletableFuture<Integer> cleanupGuildUsers(Guild guild) {
         return CompletableFuture.supplyAsync(() -> {
-            var savedIds = guildData.guildUserIds(guild);
+            var savedIds = guilds.guild(guild).userIds();
             var memberIds = guild.loadMembers().get()
                     .stream()
-                    .map(ISnowflake::getIdLong)
-                    .collect(Collectors.toList());
-            var collect = savedIds.stream().filter(id -> !memberIds.contains(id)).collect(Collectors.toList());
-            for (var id : collect) gdprData.executeRemovalTask(new RemovalTask(-1, guild.getIdLong(), id));
+                    .map(ISnowflake::getIdLong).toList();
+            var collect = savedIds.stream().filter(id -> !memberIds.contains(id)).toList();
+            for (var id : collect) new RemovalTask(gdpr, -1, guild.getIdLong(), id);
```

Comment:
The cleanup task also needs to be executed :D
```suggestion
            for (var id : collect) new RemovalTask(gdpr, -1, guild.getIdLong(), id).executeRemovalTask();
```

Reply:
I created another static method to create and directly execute removal tasks.